packages/components/src/custom-select-control-v2/custom-select-item.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import type { CustomSelectItemProps } from './types';
+import type { WordPressComponentProps } from '../context';
+import * as Styled from './styles';
+import { CustomSelectContext } from './custom-select';
+
+export function CustomSelectItem( {
+	children,
+	...props
+}: WordPressComponentProps< CustomSelectItemProps, 'div', false > ) {
+	const customSelectContext = useContext( CustomSelectContext );
+	return (
+		<Styled.CustomSelectItem
+			store={ customSelectContext?.store }
+			{ ...props }
+		>
+			{ children ?? props.value }
+			<Styled.SelectedItemCheckmark
+				// set initial size of check to prevent inline overrides
+				// due to legacy style options
+				style={ {
+					fontSize: 'initial',
+				} }
```

Comment:
It didn't affect the checkmark in the legacy verison because there was a hard size (`24px`) whereas in ariakit's it's `1em`. But this was resolved when updating to the `wordpress/icon` checkmark. 😄